examples/keyvalue-inmemory/keyvalue.go

```diff
@@ -14,16 +15,17 @@ var (
 	errInvalidDataType = store.NewError_Other("invalid data type stored in map")
 )
 
-type Provider struct {
-	sync.Map
+type Handler struct {
+	provider      *sdk.WasmcloudProvider
+	inMemoryStore sync.Map
```

Comment:
I think just keeping this field unnamed as `sync.Map` would be more "idiomatic" and easier to use.
Alternatively, it could be named `map` or `store`, if it's important not to make it public (although `provider` is already taking care of that) something along those lines. `isMemoryStore` field certainly does not look like "idiomatic" Go

Reply:
Agreed re: naming, I embedded both of the fields and it looks pretty good 👍🏻